### PR TITLE
Pipeline updates (including strict syntax compliance)

### PR DESCRIPTION
**Pull request by ewels**
Update the pipeline modules, subworkflows. Also update nf-test and bump the minimum Nextflow version so that we can run with strict syntax (`NXF_SYNTAX_PARSER=v2`).

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

File: nextflow.config [hunk 1]
```diff
@@ -265,7 +265,7 @@ manifest {
     description     = """An nf-core demo pipeline"""
     mainScript      = 'main.nf'
     defaultBranch   = 'master'
-    nextflowVersion = '!>=26.10.0'
```
Comment (mashehu) [inline nextflow.config]: Image

**Comment by vdauwera** [comment]
I'm probably doing something wrong but when I try to run this I get:

```
 N E X T F L O W   ~  version 25.10.2

Launching `./main.nf` [extravagant_varahamihira] DSL2 - revision: bf50b4675c

A workflow named 'UTILS_NFSCHEMA_PLUGIN' is already defined or included in script: /workspaces/demo/subworkflows/local/utils_nfcore_demo_pipeline/main.nf
```

File: .github/workflows/nf-test.yml [hunk 2]
```diff
@@ -78,7 +78,7 @@ jobs:
           - isMain: false
             profile: "singularity"
         NXF_VER:
-          - "25.04.0"
+          - "25.10.0"
```
Comment (vdauwera) [inline .github/workflows/nf-test.yml]: shouldn't this be "25.10.2" ?

File: workflows/demo.nf [hunk 3]
```diff
@@ -31,7 +31,7 @@ workflow DEMO {
     FASTQC (
         ch_samplesheet
     )
-    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]})
+    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{ _meta, zip -> zip })
```
Comment (vdauwera) [inline workflows/demo.nf:34]: Is the underscore in `_meta` intentional? apologies if this is a n00b question

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,37 @@
...
+- [Matthias Hörtenhuber](https://github.com/mashehu)
+- [Maxime Garcia](https://github.com/maxulysse)
+
+### Changed
+
+- [#49](https://github.com/nf-core/demo/pull/49) - Template update for nf-core/tools v3.4.1 and v3.5.1
+- Updated nf-core modules and subworkflows
+- Bump nf-schema and minimum Nextflow version
+- Use topic channels for version handling
+- Updated
```
Comment (vdauwera) [inline CHANGELOG.md]: Could we edit the automated changelog to remove uninformative lines?

File: nextflow.config [hunk 5]
```diff
@@ -263,47 +263,19 @@ manifest {
     description     = """An nf-core demo pipeline"""
     mainScript      = 'main.nf'
     defaultBranch   = 'master'
-    nextflowVersion = '!>=25.04.0'
-    version         = '1.0.2'
+    nextflowVersion = '!>=25.10.0'
```
Comment (vdauwera) [inline nextflow.config]: "25.10.2" ?

File: README.md [hunk 6]
```diff
@@ -10,7 +10,7 @@
 [![GitHub Actions Linting Status](https://github.com/nf-core/demo/actions/workflows/linting.yml/badge.svg)](https://github.com/nf-core/demo/actions/workflows/linting.yml)[![AWS CI](https://img.shields.io/badge/CI%20tests-full%20size-FF9900?labelColor=000000&logo=Amazon%20AWS)](https://nf-co.re/demo/results)[![Cite with Zenodo](http://img.shields.io/badge/DOI-10.5281/zenodo.12192442-1073c8?labelColor=000000)](https://doi.org/10.5281/zenodo.12192442)
 [![nf-test](https://img.shields.io/badge/unit_tests-nf--test-337ab7.svg)](https://www.nf-test.com)
 
-[![Nextflow](https://img.shields.io/badge/version-%E2%89%A525.04.0-green?style=flat&logo=nextflow&logoColor=white&color=%230DC09D&link=https%3A%2F%2Fnextflow.io)](https://www.nextflow.io/)
+[![Nextflow](https://img.shields.io/badge/version-%E2%89%A525.10.0-green?style=flat&logo=nextflow&logoColor=white&color=%230DC09D&link=https%3A%2F%2Fnextflow.io)](https://www.nextflow.io/)
```
Comment (vdauwera) [inline README.md]: "25.10.2" ?

**Review by vdauwera — DISMISSED** [review]

**Comment by vdauwera** [comment]
> I'm probably doing something wrong but when I try to run this I get:
> 
> ```
>  N E X T F L O W   ~  version 25.10.2
> 
> Launching `./main.nf` [extravagant_varahamihira] DSL2 - revision: bf50b4675c
> 
> A workflow named 'UTILS_NFSCHEMA_PLUGIN' is already defined or included in script: /workspaces/demo/subworkflows/local/utils_nfcore_demo_pipeline/main.nf
> ```

Tried again in a clean environment with NXF_VER="25.10.2". The pipeline runs fine if I set the parser to v1, but produces the error above if I set the parser to v2.

```
training -> export NXF_SYNTAX_PARSER=v1
training -> nextflow run nf-core/demo -profile docker,test --outdir whatever -r strict-syntax
Nextflow 25.10.3 is available - Please consider updating your version to it

 N E X T F L O W   ~  version 25.10.2

Downloading plugin nf-schema@2.6.1
Launching `https://github.com/nf-core/demo` [lethal_newton] DSL2 - revision: 0fde217be9 [strict-syntax]


------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/demo 1.1.0
------------------------------------------------------

Input/output options
  input                     : https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv
  outdir                    : whatever

Institutional config options
  config_profile_name       : Test profile
  config_profile_description: Minimal test dataset to check pipeline function

Generic options
  trace_report_suffix       : 2026-01-26_13-55-04

Core Nextflow options
  revision                  : strict-syntax
  runName                   : lethal_newton
  containerEngine           : docker
  launchDir                 : /workspaces/training
  workDir                   : /workspaces/training/work
  projectDir                : /workspaces/.nextflow/assets/nf-core/demo
  userName                  : root
  profile                   : docker,test
  configFiles               : /workspaces/.nextflow/assets/nf-core/demo/nextflow.config

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------

* The pipeline
    https://doi.org/10.5281/zenodo.12192442

* The nf-core framework
    https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
    https://github.com/nf-core/demo/blob/master/CITATIONS.md

executor >  local (7)
[86/715f41] NFCORE_DEMO:DEMO:FASTQC (SAMPLE3_SE)     [100%] 3 of 3 ✔
[87/5eae83] NFCORE_DEMO:DEMO:SEQTK_TRIM (SAMPLE3_SE) [100%] 3 of 3 ✔
[82/627bc3] NFCORE_DEMO:DEMO:MULTIQC                 [100%] 1 of 1 ✔
-[nf-core/demo] Pipeline completed successfully-

training -> export NXF_SYNTAX_PARSER=v2
training -> nextflow run nf-core/demo -profile docker,test --outdir whatever -r strict-syntax
Nextflow 25.10.3 is available - Please consider updating your version to it

 N E X T F L O W   ~  version 25.10.2

Launching `https://github.com/nf-core/demo` [big_shannon] DSL2 - revision: 0fde217be9 [strict-syntax]

A workflow named 'UTILS_NFSCHEMA_PLUGIN' is already defined or included in script: /workspaces/.nextflow/assets/nf-core/demo/subworkflows/local/utils_nfcore_demo_pipeline/main.nf
```

**Comment by ewels** [comment]
@vdauwera this is solved in Nextflow `master` already (which is why I didn't spot it locally) and due to be backported to 25.10.x - see https://github.com/nextflow-io/nextflow/pull/6539

**Comment by ewels** [comment]
Ah good timing, see https://github.com/nextflow-io/nextflow/issues/6533#issuecomment-3799846760

> I ended up backporting it, so it should be in 25.10.4

@bentsherman can read minds 😆

**Comment by ewels** [comment]
This is the same issue that I hit last week in @nf-core proteinfamilies: https://github.com/nf-core/proteinfamilies/pull/148#issuecomment-3789287722

> This seems to be a Nextflow version thing:
>
> :white_check_mark:  `25.10.2` normal
> :x: `25.10.2` with `NXF_SYNTAX_PARSER=v2`
> :white_check_mark: `25.12.0-edge`
> :white_check_mark: `25.12.0-edge` with `NXF_SYNTAX_PARSER=v2`

**Comment by ewels** [comment]
So @vdauwera I think that basically we're fine. Two common scenarios:

* Someone is running on an older version of Nextflow (eg. 25.10) - should work as using the v1 parser by default
* Someone is running on a new version of Nextflow (eg. 26.04) - should work as using the v2 parser by default but bugfix is in place

The only problem is if someone is using an older version (before 25.10.4) and opts-in to the new strict syntax. Then it'll break.

Are we ok to merge + release?

File: nextflow.config (hunk 5 above)
Comment (ewels) [reply nextflow.config]: Not sure it really matters much, the minimum version is mostly to make the nf-schema plugin upgrade work - and I think that works in 25.10.0. But I've bumped all to 25.10.2 anyway 👍🏻

File: CHANGELOG.md (hunk 4 above)
Comment (ewels) [reply CHANGELOG.md]: I don't think that this was automation's fault, this was just me getting distracted halfway through editing the file 😆

File: workflows/demo.nf (hunk 3 above)
Comment (ewels) [reply workflows/demo.nf:34]: Yes - it silences a warning from `nextflow lint` about it being an unused variable.

**Comment by ewels** [comment]
> The only problem is if someone is using an older version (before 25.10.4) and opts-in to the new strict syntax. Then it'll break.

...which is exactly the case for the new Nextflow training 🙈 Ok, I think we'll have to specify the v1 parser until 25.10.4 is released.

**Comment by ewels** [comment]
Merging, see you again in dev -> master

**Comment by ewels** [comment]
See https://github.com/nf-core/demo/pull/51
